himan-scripts/rail_weather.lua

```diff
@@ -0,0 +1,137 @@
+function SumT(TMinus,TPlus,curTime,par)
+  local psum = {}
+
+  local stopStep = math.max(curTime:GetStep():Hours() + TMinus , 0)
+  local startStep = math.min(curTime:GetStep():Hours() + TPlus , 240)
+
+  -- set initial time step
+  local mytime = forecast_time(curTime:GetOriginDateTime(),time_duration(HPTimeResolution.kHourResolution,startStep))
+
+
+  while true do
+
+    if mytime:GetStep():Hours() < stopStep or mytime:GetStep():Hours() < 0 then
+      break
+    end
+
+    local p = luatool:FetchWithType(current_time, current_level, par, current_forecast_type)
+
+    if p then
+      if #psum == 0 then
+        for i=1, #p do
+         psum[i] = 0
+        end
+      end
+
+      for i=1, #p do
+        psum[i] = psum[i] + p[i] 
+      end
+    end
+
+    -- adjust time
+    mytime:GetValidDateTime():Adjust(HPTimeResolution.kHourResolution, -1)
```

Comment:
This is ok but it will waste a lot of effort when we know that 144h onwards data is present only every 6hrs. Means more log spam of "data not found" :)

Reply:
This needs to be run in combination with transformer time interpolation to mimic the behavior of the smartmet functions.